relay-sampling/src/config.rs

```diff
@@ -81,14 +83,33 @@ impl SamplingRule {
         self.condition.supported() && self.ty != RuleType::Unsupported
     }
 
+    /// Returns `true` if rule is a reservoir rule.
+    pub fn is_reservoir(&self) -> bool {
+        matches!(&self.sampling_value, &SamplingValue::Reservoir { .. })
+    }
+
     /// Returns the sample rate if the rule is active.
-    pub fn sample_rate(&self, now: DateTime<Utc>) -> Option<SamplingValue> {
+    pub fn sample_rate(
+        &self,
+        now: DateTime<Utc>,
+        reservoir: Option<&Arc<ReservoirEvaluator>>,
+    ) -> Option<SamplingValue> {
         if !self.time_range.contains(now) {
             // Return None if rule is inactive.
             return None;
         }
 
-        let sampling_base_value = self.sampling_value.value();
+        let sampling_base_value = match self.sampling_value {
+            SamplingValue::SampleRate { value } => value,
+            SamplingValue::Factor { value } => value,
+            SamplingValue::Reservoir { limit } => {
```

Comment:
ugly: we return SamplingValue::Reservoir{limit}, even though the limit no longer has any meaning passed this point. It was the easiest solution, will refactor in the future.